code/modules/mob/living/living_life.dm

```diff
@@ -92,6 +96,19 @@
 	if(stat != DEAD)
 		return TRUE
 
+/mob/living/proc/handle_light_adjustment(obj/screen/fullscreen/see_through_darkness/S)
+	var/current = S.alpha / 255		//Our current adjustedness.
+	var/brightness = 0				//We'll assume it's superdark if we can't find something else..
+	if(isturf(loc))
+		var/turf/T = loc			//Will be true 99% of the time, thus avoiding the whole elif chain.
+		brightness = T.get_lumcount()
```

Comment:
what happens if you're in a locker here?

Reply:
Good catch, I lazily copypasted the old code here, will make this work better

Reply:
yeah, I would double check that light still works properly when you're inside an object like a box, closet, mob, are a slaughter demon, pulse demon, etc. If the lights turn off/on, or flicker, does it need to adjust aswell?

Reply:
Quick update: using get_turf seems to work fine luckily, but the issue lies more in the fact that mesons work weird, making the meson tiles brightly lit when you are standing in the light, and making it adjust to the darkness when you yourself move into it, I'm looking into why this works as it does.


https://github.com/ParadiseSS13/Paradise/assets/108773801/7db48960-bf99-417a-a127-fa3409757c58